pkg/k8s/factory_functions.go

```diff
@@ -229,3 +230,139 @@ func EqualV1Namespace(ns1, ns2 *v1.Namespace) bool {
 	return ns1.Name == ns2.Name &&
 		comparator.MapStringEquals(ns1.GetLabels(), ns2.GetLabels())
 }
+
+// ConvertToNetworkPolicy converts a *networkingv1.NetworkPolicy into a
+// *types.NetworkPolicy
+func ConvertToNetworkPolicy(obj interface{}) interface{} {
+	netPol, ok := obj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		return nil
+	}
+	// TODO check which fields we really need
+	return &types.NetworkPolicy{
+		NetworkPolicy: netPol,
+	}
+}
+
+// ConvertToK8sService converts a *v1.Service into a *types.Service
+func ConvertToK8sService(obj interface{}) interface{} {
+	service, ok := obj.(*v1.Service)
+	if !ok {
+		return nil
+	}
+	// TODO check which fields we really need
+	return &types.Service{
+		Service: service,
+	}
+}
+
+// ConvertToK8sEndpoints converts a *v1.Endpoints into a *types.Endpoints
+func ConvertToK8sEndpoints(obj interface{}) interface{} {
+	endpoints, ok := obj.(*v1.Endpoints)
+	if !ok {
+		return nil
+	}
+	// TODO check which fields we really need
+	return &types.Endpoints{
+		Endpoints: endpoints,
+	}
+}
+
+// ConvertToIngress converts a *v1beta1.Ingress into a *v1beta1.Ingress
+func ConvertToIngress(obj interface{}) interface{} {
+	ingress, ok := obj.(*v1beta1.Ingress)
+	if !ok {
+		return nil
+	}
+	// TODO check which fields we really need
+	return &types.Ingress{
+		Ingress: ingress,
+	}
+}
+
+// ConvertToCNPWithStatus converts a *cilium_v2.CiliumNetworkPolicy into a
+// *types.SlimCNP
+func ConvertToCNPWithStatus(obj interface{}) interface{} {
+	cnp, ok := obj.(*cilium_v2.CiliumNetworkPolicy)
+	if !ok {
+		return nil
+	}
+	slimCNP := &types.SlimCNP{
+		CiliumNetworkPolicy: cnp,
+	}
+	return slimCNP
+}
+
+// ConvertToCNP converts a *cilium_v2.CiliumNetworkPolicy into a *types.SlimCNP
+// without the Status field of the given CNP.
+// WARNING calling this function will set *all* fields of the given CNP as
+// empty.
+func ConvertToCNP(obj interface{}) interface{} {
+	cnp, ok := obj.(*cilium_v2.CiliumNetworkPolicy)
+	if !ok {
+		return nil
+	}
+	slimCNP := &types.SlimCNP{
+		CiliumNetworkPolicy: &cilium_v2.CiliumNetworkPolicy{
+			TypeMeta:   cnp.TypeMeta,
+			ObjectMeta: cnp.ObjectMeta,
+			Spec:       cnp.Spec,
+			Specs:      cnp.Specs,
+		},
+	}
+	*cnp = cilium_v2.CiliumNetworkPolicy{}
+	return slimCNP
+}
+
+// ConvertToPod converts a *v1.Pod into a *types.Pod.
+// WARNING calling this function will set *all* fields of the given Pod as
+// empty.
+func ConvertToPod(obj interface{}) interface{} {
```

Comment:
Can you explain why the fields need to be set to empty? Is this so the old state can be GC'd?